src/Classes/PolygonMask.ts

```diff
@@ -0,0 +1,222 @@
+"use strict";
+
+import {Container} from "./Container";
+import {ICoordinates} from "../Interfaces/ICoordinates";
+import {PolygonMaskType} from "../Enums/PolygonMaskType";
+import {Particle} from "./Particle";
+
+export class PolygonMask {
+    public redrawTimeout?: number;
+    public raw?: number[][];
+    public svg?: SVGSVGElement;
+    public path?: SVGPathElement;
+
+    private readonly container: Container;
+    private width: number;
+    private height: number;
+    private offset?: ICoordinates;
+
+    constructor(container: Container) {
+        this.container = container;
+        this.width = 0;
+        this.height = 0;
+    }
+
+    public checkInsidePolygon(position: ICoordinates | undefined | null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'checkInsidePolygon' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5042590)